website/docs/r/wafregional_web_acl_association.html.markdown

```diff
@@ -0,0 +1,94 @@
+---
+layout: "aws"
+page_title: "AWS: aws_wafregional_web_acl_association"
+sidebar_current: "docs-aws-resource-wafregional-web-acl-association"
+description: |-
+  Provides a resource to create an association between a WAF Regional WebACL and Application Load Balancer.
+---
+
+# aws\_wafregional\_web\_acl\_association
+
+Provides a resource to create an association between a WAF Regional WebACL and Application Load Balancer.
+
+-> **Note:** An Application Load Balancer can only be associated with one WAF Regional WebACL.
+
+## Example Usage
+
+```hcl
+resource "aws_wafregional_ipset" "ipset" {
+  name = "tfIPSet"
+  
+  ip_set_descriptors {
+    type  = "IPV4"
+    value = "192.0.7.0/24"
+  }
+}
+
+resource "aws_wafregional_rule" "wafrule" {
+  depends_on  = ["aws_wafregional_ipset.ipset"]
+  name        = "tfWAFRule"
+  metric_name = "tfWAFRule"
+  
+  predicates {
+    data_id = "${aws_wafregional_ipset.ipset.id}"
+    negated = false
+    type    = "IPMatch"
+  }
+}
+
+resource "aws_wafregional_web_acl" "wafacl" {
+  depends_on  = ["aws_wafregional_ipset.ipset", "aws_wafregional_rule.wafrule"]
+  name        = "tfWebACL"
+  metric_name = "tfWebACL"
+  
+  default_action {
+    type = "ALLOW"
+  }
+  
+  rules {
+    action {
+       type = "BLOCK"
+    }
+    
+    priority = 1
+    rule_id  = "${aws_wafregional_rule.wafrule.id}"
+  }
+}
+
+resource "aws_vpc" "main" {
+	cidr_block = "10.1.0.0/16"
+}
+
+resource "aws_subnet" "foo" {
+	vpc_id     = "${aws_vpc.main.id}"
+	cidr_block = "10.1.1.0/24"
+}
+
+resource "aws_subnet" "bar" {
+	vpc_id     = "${aws_vpc.main.id}"
+	cidr_block = "10.1.2.0/24"
+}
+
+resource "aws_alb" "alb" {
+    subnets = ["${aws_subnet.foo.id}", "${aws_subnet.bar.id}"]
```

Comment:
We'll either need to make this ALB internal, so it doesn't need IGW, or add an IGW to the VPC.
Also we'll need to use `aws_availability_zones` data source to ensure that each subnet is created in a different AZ, otherwise we might end up with both subnets in the same AZ, which will effectively cause the ALB creation to fail (it requires subnets in two different AZs).